roles/edpm_reboot/tasks/main.yaml

```diff
@@ -0,0 +1,69 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Make sure yum-utils is installed
+  ansible.builtin.package:
+    name: yum-utils
+
+- name: Check if reboot is required with needs-restarting
+  ansible.builtin.command: needs-restarting -r
+  register: needs_restarting_output
+  changed_when: false
+
+- name: Print return information from needs-restarting
+  ansible.builtin.debug:
+    var: needs_restarting_output.stdout
+
+- name: Create reboot file for needs-restarting
+  when: _no_reboot_message not in needs_restarting_output.stdout
```

Comment:
Since you already record results of the task, you can use `rc` to get process return code and decide based on that. This will allow you to drop the variable.